src/rm/rm.rs

```diff
@@ -129,7 +129,7 @@ fn remove(files: Vec<String>, force: bool, interactive: InteractiveMode, one_fs:
         let filename = &filename[..];
         let file = Path::new(filename);
         if file.exists() {
-            if file.is_dir() {
+            if file.symlink_metadata().unwrap().is_dir() {
```

Comment:
Done, but I couldn't test this code path. I think if file.symlink_metadata() fails, file.exists() above would also fail at least on linux (because they are stat and lstat on linux).